packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -667,6 +708,148 @@ class SyncWatcher extends BaseWatcher {
+    const destinationChain = this.chainIdToSlug(destinationChainId)
+    const destinationWatcher = this.getSiblingWatcherByChainSlug(destinationChain)
+    if (!destinationWatcher) {
+      throw new Error(`no destination watcher for ${destinationChain}`)
+    }
+    const destinationBridge = destinationWatcher.bridge
+    let availableCredit = await destinationBridge.getAvailableCredit()
+    const subtractPendingAmount = destinationChain === Chain.Ethereum && bondableChains.includes(sourceChain)
+    if (subtractPendingAmount) {
+      let pendingAmounts = BigNumber.from(0)
+      for (const chain of bondableChains) {
+        const watcher = this.getSiblingWatcherByChainSlug(chain)
+        if (!watcher) {
+          continue
+        }
+        const bridge = watcher.bridge as L2Bridge
+        const pendingAmount = await bridge.getPendingAmountForChainId(destinationChainId)
+        pendingAmounts = pendingAmounts.add(pendingAmount)
+      }
+      availableCredit = availableCredit.sub(pendingAmounts)
+    }
+
+    if (availableCredit.lt(0)) {
+      return BigNumber.from(0)
+    }
+
+    return availableCredit
+  }
+
+  private async updateAvailableCredit (destinationChainId: number) {
+    const availableCredit = await this.calculateAvailableCredit(destinationChainId)
+    const destinationChain = this.chainIdToSlug(destinationChainId)
+    this.lastAvailableCredit[destinationChain] = availableCredit
+  }
+
+  private async syncAvailableCredit () {
+    const chains = await this.bridge.getChainIds()
+    for (const destinationChainId of chains) {
+      const sourceChain = this.chainSlug
+      const destinationChain = this.chainIdToSlug(destinationChainId)
+      const shouldSkip = (
+        sourceChain === Chain.Ethereum ||
+        sourceChain === destinationChain ||
+        !this.hasSiblingWatcher(destinationChainId)
+      )
+      if (shouldSkip) {
+        continue
+      }
+      await this.updateAvailableCredit(destinationChainId)
+      const lastAvailableCredit = await this.getLastAvailableCredit(destinationChainId)
+      this.logger.debug(`lastAvailableCredit (${this.tokenSymbol} ${sourceChain}→${destinationChain}): ${this.bridge.formatUnits(lastAvailableCredit)}`)
+    }
+  }
+
+  public async getLastAvailableCredit (destinationChainId: number, amount: BigNumber = BigNumber.from(0)) {
+    const sourceChain = this.chainSlug
+    const destinationChain = this.chainIdToSlug(destinationChainId)
+    let availableCredit = this.lastAvailableCredit[destinationChain]
+    if (!availableCredit) {
+      return BigNumber.from(0)
+    }
+    const subtractAmount = destinationChain === Chain.Ethereum && bondableChains.includes(sourceChain)
+    if (subtractAmount) {
+      availableCredit = availableCredit.sub(amount)
+    }
+
+    if (availableCredit.lt(0)) {
+      return BigNumber.from(0)
+    }
+
+    return availableCredit
+  }
+
+  public async calculateRecentCommitPendingAmounts (destinationChainId: number) {
```

Comment:
Since we use the word "pending" to describe un-committed amounts, I would suggest we use a different name for all of these functions that have to do with the "in-flight" commits.

My vote would be "unsetCommits", so `calculateUnsetCommitAmounts`, for example.

Reply:
I like the unsetCommits terminology, will update